### PR TITLE
Better exception when JScript.NET is targeted

### PR DESCRIPTION
**Pull request by pjanotti**
Generates the same exception for the other languages that are not
supported: ArgumentOutOfRangeException.

No tests are being added for these exceptions because failure to
generate the file keeps the file locked, so that would leave files with
size 0 on the test box.

### CODE REVIEW

**Comment by pjanotti** [comment]
Just to make clear that we could workaround the file issue to have a test, something like:
```c#
                GC.Collect(2);
                GC.WaitForPendingFinalizers();
```
should do it, but it seems a bit too much for such case.

File: src/System.Management/src/System/Management/WMIGenerator.cs [hunk 1]
```diff
@@ -4861,7 +4861,12 @@ bool InitializeCodeGenerator(CodeLanguage lang)
                         strProvider = "Visual Basic.";
                         cp = new VBCodeProvider();
                         break;
-            
+
+                    case CodeLanguage.JScript:
+                        strProvider = "JScript.NET.";
+                        bSucceeded = false; // JScriptCodeProvider does not exist on CoreFx
+                        break;
```
Comment (stephentoub) [inline src/System.Management/src/System/Management/WMIGenerator.cs]: There a bunch of code for other `bSucceeded = false;` cases for other languages below, e.g. VJSharp, Mcpp.  What's the reason they need all that assembly-related code?  If it's not necessary, can/should it be removed?  If it is necessary, is similar code necessary here?

Comment (pjanotti) [reply src/System.Management/src/System/Management/WMIGenerator.cs]: So far we have adopted the policy of keeping this code as close as possible to the desktop code, that's why those are left there. Notice that `System.Management` is no longer being developed (https://msdn.microsoft.com/en-us/library/aa393964(v=vs.85).aspx), this work is now done under a different namespace.

The assemblies for Mcpp and VJSharp are not shipped with desktop but in principle an app can ship them and try to generate the code, that said I didn't try it myself (I still can see someone using Mcpp, but VJSharp just recently got out of support window: 
> Support for Visual J# 2.0 Redistributable Package Second Edition ends on 10/10/2017. For more information, see https://msdn.microsoft.com/en-us/vstudio/bb188593.

Comment (stephentoub) [reply src/System.Management/src/System/Management/WMIGenerator.cs]: Ok.  Do we need a test for this change?

Comment (pjanotti) [reply src/System.Management/src/System/Management/WMIGenerator.cs]: I added the test, it indeed needed the GC snippet above to not leave files on the disk. Notice that currently we don't run these tests against netfx on CI but I do it manually whenever I change anything.

**Pull request merged**